pointcloud/basic_octree_utils_test.go

```diff
@@ -327,3 +330,63 @@ func printBasicOctree(logger golog.Logger, bOct *BasicOctree, s string) {
 		printBasicOctree(logger, v, s+"-+-")
 	}
 }
+
+// Test the functionalities involved with converting a pointcloud into a basic octree.
+func TestBasicOctreeCollision(t *testing.T) {
+	startPC, err := makeFullPointCloudFromArtifact(
+		t,
+		"slam/example_cartographer_outputs/viam-office-02-22-1/pointcloud/pointcloud_0.pcd",
+		BasicType,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	center := getCenterFromPcMetaData(startPC.MetaData())
+	maxSideLength := getMaxSideLengthFromPcMetaData(startPC.MetaData())
+
+	basicOct, err := NewBasicOctree(center, maxSideLength)
+	test.That(t, err, test.ShouldBeNil)
+
+	startPC.Iterate(0, 0, func(p r3.Vector, d Data) bool {
+		_, _, blueProb := d.RGB255()
+		d.SetValue(int(blueProb))
+		if err = basicOct.Set(p, d); err != nil {
+			return false
+		}
+		return true
+	})
+
+	test.That(t, startPC.Size(), test.ShouldEqual, basicOct.Size())
+
+	// create a non-colliding obstacle far away from any octree point
+	far, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{1, 2, 3}, "far")
+	test.That(t, err, test.ShouldBeNil)
+	// create a non-colliding obstacle near an octree point
+	near, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{-2443, 0, 3855}), r3.Vector{1, 2, 3}, "near")
```

Comment:
[nit] maybe write this test in a way such that this doesn't break if the artifact point cloud changes? Also question: how did you verify that these geometries do what you are testing for?

Reply:
How often do we change the artifacts upon which we are performing tests? This should be stable, and we would want to know if these change by failing the test.

The geometries were selected by opening the point cloud in question, looking at the points manually, and constructing geometries around them by hand such that they would behave as desired.

Reply:
That seems ok.  I was just probing if it was possible to construct this test programmatically instead of hard coding things, since it's not guaranteed that changing the artifact would be testing the same things.  

But it isn't obvious to me how you would do that so maybe we should just proceed down the easy path